src/torchcodec/decoders/_decoder_utils.py

```diff
@@ -18,18 +19,32 @@
 
 
 def create_decoder(
-    *, source: Union[str, Path, bytes, Tensor], seek_mode: str
+    *,
+    source: Union[str, Path, io.RawIOBase, io.BufferedReader, bytes, Tensor],
+    seek_mode: str,
 ) -> Tensor:
     if isinstance(source, str):
         return core.create_from_file(source, seek_mode)
     elif isinstance(source, Path):
         return core.create_from_file(str(source), seek_mode)
+    elif isinstance(source, io.RawIOBase) or isinstance(source, io.BufferedReader):
+        return core.create_from_file_like(source, seek_mode)
     elif isinstance(source, bytes):
         return core.create_from_bytes(source, seek_mode)
     elif isinstance(source, Tensor):
         return core.create_from_tensor(source, seek_mode)
+    elif isinstance(source, io.TextIOBase):
+        raise TypeError(
+            "source is of type io.TextIOBase; did you forget to specify binary reading?"
```

Comment:
Nit: it's not obvious what an `io.TextIOBase` is (I have no idea). IMO we should just guess something like
```suggestion
            "Looks like you specified the source as open(..., 'r')? Try with 'rb' for binary reading?"
```